packages/design-systems/src/components/composite/Dialog.tsx

```diff
@@ -0,0 +1,104 @@
+import { ReactNode } from "react";
+import {
+  Dialog as ArkDialog,
+  DialogProps as ArkDialogProps,
+  Portal,
+} from "@ark-ui/react";
+import {
+  dialog,
+  DialogVariantProps,
+} from "@tailor-platform/styled-system/recipes";
+import { X as XIcon } from "lucide-react";
+import { Button } from "../Button";
+import { IconButton } from "../IconButton";
+import { Stack } from "../patterns/Stack";
+
+type DialogContentProps = {
+  buttonText: string;
+  title: string;
+  description: ReactNode;
+  cancelText?: string;
+  confirmText?: string;
+  isOpen: boolean;
+  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
```

Comment:
ここ必ずしも `React.Dispatch` が来るとは限らないので、以下のようにgenericな関数シグニチャにしてもらいたいです。

```suggestion
  setIsOpen: (isOpen: boolean) => void;
```

Reply:
あー確かに
了解しました！修正します！